cmd/argocd/commands/app.go

```diff
@@ -110,6 +116,7 @@ func NewApplicationAddCommand(clientOpts *argocdclient.ClientOptions) *cobra.Com
 	command.Flags().StringVar(&env, "env", "", "Application environment to monitor")
 	command.Flags().StringVar(&destServer, "dest-server", "", "K8s cluster URL (overrides the server URL specified in the ksonnet app.yaml)")
 	command.Flags().StringVar(&destNamespace, "dest-namespace", "", "K8s target namespace (overrides the namespace specified in the ksonnet app.yaml)")
+	command.Flags().StringVar(&syncPolicy, "sync-policy", "", "K8s sync policy (overrides the sync policy specified in the ksonnet app.yaml)")
```

Comment:
Description is not quite right. Ksonnet doesn't understand sync policies. You can just say "Synchronization policy for application (e.g. Always)"